sys/cheri/cheric.h

```diff
@@ -76,6 +76,28 @@
 #define	cheri_setbounds(x, y)	__builtin_cheri_bounds_set((x), (y))
 #define	cheri_setboundsexact(x, y)	__builtin_cheri_bounds_set_exact((x), (y))
 
+/* Compare capabilities including bounds and perms etc. */
+#define cheri_equal_exact(x, y) __builtin_cheri_equal_exact(x, y)
+
+/*
+ * Return whether the two pointers are equal, including capability metadata if
+ * in purecap mode.
+ */
+#ifdef __cplusplus
+static inline bool
+#else
+static inline _Bool
+#endif
+cheri_pointers_equal(void *x, void *y) {
```

Comment:
I think something like cheri_ptr_equal_exact would be more in keeping with the naming here?

Reply:
OK